Sources/ClientRuntime/Networking/AwsChunked/AWSChunkedUtil.swift

```diff
@@ -0,0 +1,113 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import AwsCommonRuntimeKit
+
+extension SigningConfig {
+    public var useAwsChunkedEncoding: Bool {
+        switch self.signedBodyValue {
+        case .streamingSha256Payload, .streamingSha256PayloadTrailer, .streamingUnSignedPayloadTrailer:
+            return true
+        default:
+            return false
+        }
+    }
+}
+
+extension SdkHttpRequestBuilder {
+    public func setAwsChunkedHeaders() throws {
+
+        // Check if self.body is of the case ByteStream.stream(let stream)
+        if case .stream(let stream) = self.body {
+            // Set the common headers for AWS-chunked encoding
+            self.withHeader(name: "Content-Encoding", value: "aws-chunked")
+            self.withHeader(name: "Transfer-Encoding", value: "chunked")
+            guard let decodedContentLength = stream.length else {
+                throw ClientError.dataNotFound("Cannot use aws-chunked encoding with an unknown stream length!")
+            }
+            self.withHeader(name: "X-Amz-Decoded-Content-Length", value: String(decodedContentLength))
+        } else {
+            throw ClientError.dataNotFound("aws-chunked encoding requires a streaming payload!")
+        }
+    }
+
+    public func setAwsChunkedBody(
```

Comment:
This method now just wraps the body stream in a AWSChunkedStream when the request is chunked.